Source/Core/Monomorphization.cs

```diff
@@ -1293,11 +1297,20 @@ class MonomorphizationVisitor : VarDeclOnceStandardVisitor
     private HashSet<TypeCtorDecl> visitedTypeCtorDecls;
     private HashSet<Function> visitedFunctions;
     private Dictionary<Procedure, Implementation> procToImpl;
+    public readonly HashSet<Function> originalFunctions;
+    public readonly HashSet<Constant> originalConstants;
+    // Note that old axioms refer to axioms of the original program which might have been updated in-place during monomorphization.
```

Comment:
Are you sure old axioms are updated in-place? If I remember correctly, each axiom could be instantiated multiple times so the instantiation is done on a duplicate version of the axiom which is modified in-place. The original axiom stays as is to facilitate the next instantiation. Could you please check?

Reply:
Yes, I checked now and old axioms get instantiated in-place after executing [this line](https://github.com/boogie-org/boogie/blob/0b1e6843c94480df8e05790f43ae4182e52e3204/Source/Core/Monomorphization.cs#L627). I think [this comment](https://github.com/boogie-org/boogie/blob/0b1e6843c94480df8e05790f43ae4182e52e3204/Source/Core/Monomorphization.cs#L1239-L1243) also confirms that the old axioms are instantiated in-place (with the help of `MonomorphizationDuplicator`).